api/test_runs.go

```diff
@@ -130,3 +140,26 @@ func LoadTestRunsForFilters(ctx context.Context, filters shared.TestRunFilter) (
 	}
 	return shared.LoadTestRunsByKeys(ctx, keys)
 }
+
+func getPRCommits(ctx context.Context, pr int) []string {
+	log := shared.GetLogger(ctx)
+	secret, err := shared.GetSecret(ctx, "github-api-token")
+	if err != nil {
+		log.Debugf("Failed to load github-api-token: %s", err.Error())
+		return nil
+	}
+	oauthClient := oauth2.NewClient(ctx, oauth2.StaticTokenSource(&oauth2.Token{
+		AccessToken: secret,
+	}))
+	githubClient := github.NewClient(oauthClient)
+	commits, _, err := githubClient.PullRequests.ListCommits(ctx, "web-platform-tests", "wpt", pr, nil)
```

Comment:
What's the default limit on number of commits returned? The last returned is the most recent and interesting, so if it falls outside the window that'd be bad. Maybe just set the limit to 100 and fail if the number of returned commits is 100 so we notice it? It's really only when people have accidentally pushed a rebase of master or something that you get that many commits.

Alternatively, paginate and get *all* commits, however many they are.

Reply:
Eh, for simplicity, I'll wait until we actually see this as a real-world problem.

Reply:
If the limit is unknown, waiting until a bug happens and is reported by a user seems too far in the iterative direction. When bugs happen, getting the fix into prod can take time even when the fix is trivial. At least let's understand the limit and fail in an explicit way so that the problem is trivially debugged when it happens.

https://developer.github.com/v3/pulls/#list-commits-on-a-pull-request claims that up to 250 commits can be listed. https://github.com/foolip/wpt/pull/5 had 68 commits, but https://api.github.com/repos/foolip/wpt/pulls/5/commits only lists 30. I don't know where the canonical documentation is, but https://octokit.github.io/rest.js/#api-Pulls-listCommits also says default 30, max 100.

I tweaked a script I had to checking on my Azure Pipelines statuses, and 30 is low enough that we have plenty of PRs exceeding it: 13499, 12026, 6467, 1242. That was for open PRs only, presumably lots of merged ones have >30 too.

Can you file an issue to follow up?

Reply:
As per https://developer.github.com/v3/#pagination, setting the limit to 100.
I'll also file an issue for paginating; we can paginate on wpt.fyi and plumb "Offset: 100" to mean "GitHub Page 2".
https://github.com/web-platform-tests/wpt.fyi/pull/792
https://github.com/web-platform-tests/wpt.fyi/issues/793